css/css-transforms/parsing/rotate-parsing-valid.html

```diff
@@ -22,18 +22,22 @@
 // If a 3d rotation is specified, the property must serialize with an axis specified.
 test_valid_value("rotate", "100 200 300 400grad");
 test_valid_value("rotate", "400grad 100 200 300", "100 200 300 400grad");
+test_valid_value("rotate", "0 0 0 400grad", "0 0 0 400grad");
 
 // If the axis is parallel with the x, y, or z axis, it must serialize as the appropriate keyword.
 test_valid_value("rotate", "x 400grad");
 test_valid_value("rotate", "400grad x", "x 400grad");
+test_valid_value("rotate", "0.5 0 0 400grad", "x 400grad");
 test_valid_value("rotate", "1 0 0 400grad", "x 400grad");
 
 test_valid_value("rotate", "y 400grad");
 test_valid_value("rotate", "400grad y", "y 400grad");
+test_valid_value("rotate", "0 0.5 0 400grad", "y 400grad");
 test_valid_value("rotate", "0 1 0 400grad", "y 400grad");
 
 test_valid_value("rotate", "z 400grad");
 test_valid_value("rotate", "400grad z", "z 400grad");
+test_valid_value("rotate", "0 0 0.5 400grad", "z 400grad");
 test_valid_value("rotate", "0 0 1 400grad", "z 400grad");
```

Comment:
These 4 tests are all invalid, I think, based on the spec edits made in w3c/csswg-drafts#6147 for w3c/csswg-drafts#3305.

Reply:
I guess I should clarify that they're invalid, I think, because the `z` should be omitted in the serialization.

Reply:
So these should not specify the `z` axis at all? This is great but should indeed be changed in a different PR which I'm happy to file after this one lands (and fix WebKit too).